### ISSUE TITLE
Ability to manually check the current heading, region or frame in a structured document

### ISSUE THREAD

**Issue by Masarczyk-Wienfluss** [nvaccess/nvda#16734]
### Is your feature request related to a problem? Please describe.
Following up on my observation from #16648 that the relation between the different semantic containers in a structured document like a website or something written in Word is not always apparent to all users I note that there is currently no way to check where in the structure we are currently located ie. to what larger part of the whole document the currently viewed piece of text or interactive control belongs. Gaining this ability would provide greater context awareness of where we are, like checking the current chapter or subchapter of a book.

### Describe the solution you'd like
Again, inspired by what is currently offered in JAWS, I would like to propose a feature where the title of the heading, aria region or frame under which the cursor is currently located is appended at the end of the title window read by pressing NVDA+T. It should be appended at the end as to not disturb the usual readout flow most users are accustomed to so ideally it would come after the name of the software is read out, ie. at the end of the entire title string.

### Describe alternatives you've considered
This same information could alternatively be callable by pressing NVDA+Tab, provided the item we are interested in is focusable. Whatever the approach, this readout could be optional for those who do not need it.

**Comment by Adriani90** [comment]
I agree this could be really helpful in certain situations, but I would rather propose to add an unassigned gesture for it or bind it to nvda+shift+t or so.

**Comment by Masarczyk-Wienfluss** [comment]
My idea was more to allow instant access to this piece of information through an already familiar technique, yet in a way that does not disrupt the usual flow which is expected to be more relevant to most users.
I think this proposal might be quite important in view of the [H80 WCAG Technique](https://www.w3.org/WAI/WCAG21/Techniques/html/H80) allowing link texts to be generic as long as the preceding heading explains the purpose of said link in a sufficient manner. This is usually employed in news galleries where each news item is a heading followed by a teaser of what one can find inside and then the "Read more" link. In such a scenario the feature would quickly allow the user to check which heading the link is referencing without having to move the focus to another place.

**Comment by TechHorseG** [comment]
I like the idea of having a 'tell me where I am in the grand scheme of things' style function, but I would agree with it having its own gesture, rather than it being added to a pre-existing function.

I am very sensitive to TMI and over-verbosity when I just want to quickly know one piece of information, and know that I would end up switching it off if I had to hear it every time I just wanted to hear the title or current object etc.

Also, if I wanted to hear the current section / heading spelled out or copied to the clipboard, then I could use its own dedicated gesture to achieve this without the TMI issue of, for example,  having to patiently wait for the title to be spelt out first.

But I do like the idea otherwise. It reminds me of a feature request I am thinking of making, where you can placemark the position you are at in a document. You can then go and recheck some previously read information such as a number or name given in a previous paragraph, or the current heading etc, and then press the same or a different key to quickly return to the previously marked location and carry on reading.

But however it is done, it would certainly be useful to have a quicker way to check information previously on the page with a minimum of disruption, rather than having to both manually find your way there and then manually find your way back.

**Comment by josephsl** [comment]
Hi,

At least for headings, one (long winded) workaround is elements list (NVDA+F7). By default it shows a list of links, so when elements list opens, press Alt+H to switch to headings view, then press Tab to look at nearest heading.

Thanks.Hi,

As for place marker feature, taek a look at Place Markers ad-on by @NVDAEs.

Thanks.

**Comment by Adriani90** [comment]
Manual checking is actually expected behavior, sighted people have to do it as well. In case of region / Headings I can understand the use case because e.g. a sighted person can imediately see the region / heading without having to move the system focus or mouse cursor.The bookmark reporting request  is valid as well, but is a separate issue and is more or less covered by the placemarker add-on  Von meinem iPhone gesendetAm 29.06.2024 um 17:03 schrieb TechHorseG ***@***.***>:﻿
I like the idea of having a 'tell me where I am in the grand scheme of things' style function, but I would agree with it having its own gesture, rather than it being added to a pre-existing function.
I am very sensitive to TMI and over-verbosity when I just want to quickly know one piece of information, and know that I would end up switching it off if I had to hear it every time I just wanted to hear the title or current object etc.
Also, if I wanted to hear the current section / heading spelled out or copied to the clipboard, then I could use its own dedicated gesture to achieve this without the TMI issue of, for example,  having to patiently wait for the title to be spelt out first.
But I do like the idea otherwise. It reminds me of a feature request I am thinking of making, where you can placemark the position you are at in a document. You can then go and recheck some previously read information such as a number or name given in a previous paragraph, or the current heading etc, and then press the same or a different key to quickly return to the previously marked location and carry on reading.
But however it is done, it would certainly be useful to have a quicker way to check information previously on the page with a minimum of disruption, rather than having to both manually find your way there and then manually find your way back.
